crates/tx5-signal-srv/src/server.rs

```diff
@@ -277,7 +300,11 @@ async fn client_task(
                     }) => {
                         // TODO - pay attention to demo config flag
                         //        right now we just always honor demos
-                        srv_hnd_read.broadcast(msg);
+                        if let Some(srv_cmd) = srv_cmd_read.upgrade() {
+                            srv_cmd.broadcast(msg);
+                        } else {
+                            break;
```

Comment:
I wouldn't say so, no... In this else case we were not able to upgrade the weak reference, so that means whoever owns us has dropped the strong handle. If they want to error when that handle is dropped, they should make a newtype with a Drop impl, no?

Reply:
I see, yes that seems sensible!